src/os/bluestore/BlueStore.cc

```diff
@@ -4519,9 +4519,8 @@ int BlueStore::fsck(bool deep)
   if (it) {
     for (it->lower_bound(string()); it->valid(); it->next()) {
       string key = it->key();
-      const char *p = key.c_str();
       uint64_t omap_head;
-      p = _key_decode_u64(p, &omap_head);
+      _key_decode_u64(key.c_str(), &omap_head);
```

Comment:
One can remove key var as well and that's even more beneficial since string is copied there